gracenet/net.go

```diff
@@ -210,13 +210,19 @@ func (n *Net) StartProcess() (int, error) {
 	}
 
 	// Extract the fds from the listeners.
+	// NOTE: gracenet not remove closed fd from active, so pass closeed fd to childprocess
 	files := make([]*os.File, len(listeners))
-	for i, l := range listeners {
-		files[i], err = l.(filer).File()
+	fcnt := 0
+	for _, l := range listeners {
+		files[fcnt], err = l.(filer).File()
 		if err != nil {
+			if isUseOfClosedError(err) {
+				continue
+			}
 			return 0, err
 		}
-		defer files[i].Close()
+		defer files[fcnt].Close()
+		fcnt++
```

Comment:
Some fds were closed would not be copied to files. so the `files` array will contain some nil objects, I have no idea about the Go standard library how to process the situation when `os.StartProcess` accepted `ProcAttr` argument which has nil objects